auditbeat/module/file_integrity/eventreader_fsnotify.go

```diff
@@ -36,22 +37,25 @@ func NewEventReader(c Config) (EventProducer, error) {
 }
 
 func (r *reader) Start(done <-chan struct{}) (<-chan Event, error) {
-	for _, p := range r.config.Paths {
-		if err := r.watcher.Add(p); err != nil {
-			if err == syscall.EMFILE {
-				r.log.Warnw("Failed to add watch (check the max number of "+
-					"open files allowed with 'ulimit -a')",
-					"file_path", p, "error", err)
-			} else {
-				r.log.Warnw("Failed to add watch", "file_path", p, "error", err)
-			}
-		}
+
+	addWatchesFirst := runtime.GOOS != "windows"
+
+	if addWatchesFirst {
```

Comment:
What's the benefit of configuring the paths before `Start`'ing in the non-windows case?

Reply:
I was afraid it could introduce problems in platforms other than Windows. That's why I created the system test, but at the end forgot to update this code.

I've pushed a new version were watches are always installed after Start().